src/database/models/collection_item.rs

```diff
@@ -0,0 +1,342 @@
+
+        let mut found_collections = Vec::new();
+        let mut remaining_strings = collection_strings
+            .iter()
+            .map(|x| x.to_string())
+            .collect::<Vec<_>>();
+
+        let mut collection_ids = collection_strings
+            .iter()
+            .flat_map(|x| parse_base62(&x.to_string()).map(|x| x as i64))
+            .collect::<Vec<_>>();
+
+        collection_ids.append(
+            &mut cmd("MGET")
+                .arg(
+                    collection_strings
+                        .iter()
+                        .map(|x| {
+                            format!(
+                                "{}:{}",
+                                COLLECTIONS_SLUGS_NAMESPACE,
+                                x.to_string().to_lowercase()
+                            )
+                        })
+                        .collect::<Vec<_>>(),
+                )
+                .query_async::<_, Vec<Option<i64>>>(&mut redis)
+                .await?
+                .into_iter()
+                .flatten()
+                .collect(),
+        );
+
+        if !collection_ids.is_empty() {
+            let collections = cmd("MGET")
+                .arg(
+                    collection_ids
+                        .iter()
+                        .map(|x| format!("{}:{}", COLLECTIONS_NAMESPACE, x))
+                        .collect::<Vec<_>>(),
+                )
+                .query_async::<_, Vec<Option<String>>>(&mut redis)
+                .await?;
+
+            for collection in collections {
+                if let Some(collection) =
+                    collection.and_then(|x| serde_json::from_str::<Collection>(&x).ok())
+                {
+                    remaining_strings.retain(|x| {
+                        &to_base62(collection.id.0 as u64) != x
+                            && collection.slug.as_ref().map(|x| x.to_lowercase())
+                                != Some(x.to_lowercase())
+                    });
+                    found_collections.push(collection);
+                    continue;
+                }
+            }
+        }
+
+        if !remaining_strings.is_empty() {
+            let collection_ids_parsed: Vec<i64> = remaining_strings
+                .iter()
+                .flat_map(|x| parse_base62(&x.to_string()).ok())
+                .map(|x| x as i64)
+                .collect();
+            let db_collections: Vec<Collection> = sqlx::query!(
+                "
+                SELECT c.id id, c.title title, c.description description,
+                c.icon_url icon_url, c.color color, c.body body, c.published published,
+                c.updated updated, c.team_id team_id, c.slug slug, c.public public,
+                ARRAY_AGG(DISTINCT m.id) filter (where m.id is not null) mods
+                FROM collections c
+                LEFT JOIN collections_mods cm ON cm.collection_id = c.id
+                LEFT JOIN mods m ON m.id = cm.mod_id
```

Comment:
is there a reason to join mods here? if its in the collection you already have the mod id